src/scipp/plt/controller1d.py

```diff
@@ -0,0 +1,16 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+from .controller import Controller
+
+
+class Controller1d(Controller):
+    """
+    Controller class for 1d plots.
+    """
+    def render(self, *args, **kwargs):
+        """
+        """
+        super().render(*args, **kwargs)
+        self._view.rescale_to_data()
+        self._view.set_axes_labels()
```

Comment:
So the `rescale_to_data` needs to be done manually because of the way we first make a line with two dummy points, and then update the x and y data of the line. Updating the line data does not make matplotlib automatically rescale.
We use this because it makes the code more generic, and it is also faster (when using a slider) than deleting the line and drawing a whole new one.

Setting the axes labels also needs to be done after the data has been updated in the plot, so after the initial figure generation and the `update()` method has been called on the view.
In the 2d plot, this is handled by checking inside `update()` if the member `self._image` is `None` or not (which it will be the first time `update()` is called).

I agree that conceptually I shouldn't have a 1d controller. I will change to a similar mechanism as in the 2d case (I avoided that initially when I was messing around with lines being added and removed with masks etc... but it will be fine  to do it like the 2d case now)

Reply:
> So the rescale_to_data needs to be done manually because of the way we first make a line with two dummy points, and then update the x and y data of the line. Updating the line data does not make matplotlib automatically rescale.

Aren't you coupling the controller to the concrete plotting lib we use? There has to be a better way that does not require the controller to know specifics about 1D matplotlib figures.